framework/math/linear_solver/linear_solver.h

```diff
@@ -78,29 +48,54 @@ class LinearSolver
   void SetKSPSolveSuppressionFlag(bool flag) { suppress_kspsolve_ = flag; }
   bool GetKSPSolveSuppressionFlag() const { return suppress_kspsolve_; }
 
+  /**
+   * Set up the linaer solver
+   */
+  virtual void Setup();
+
+  /**
+   * Solve the system
+   */
+  virtual void Solve();
+
 protected:
+  bool IsSystemSet() const { return system_set_; }
   virtual void PreSetupCallback();
   virtual void SetOptions();
   virtual void SetSolverContext();
   virtual void SetConvergenceTest();
   virtual void SetMonitor();
   virtual void SetPreconditioner();
-
   virtual void SetSystemSize() = 0;
   virtual void SetSystem() = 0;
   virtual void PostSetupCallback();
-
-public:
-  virtual void Setup();
-
-protected:
   virtual void PreSolveCallback();
   virtual void SetInitialGuess() = 0;
   virtual void SetRHS() = 0;
   virtual void PostSolveCallback();
 
-public:
-  virtual void Solve();
+  /// Solver name
+  const std::string solver_name_;
+  const std::string iterative_method_;
+
+  LinSolveContextPtr context_ptr_ = nullptr;
+
+  /// Matrix
+  Mat A_;
+  /// Right-hand side
+  Vec b_;
+  /// Solution vector
+  Vec x_;
```

Comment:
We could rename these to `matrix_`, `rhs_` and `sol_` (or `solution_`) to both meet syntax (`snake_case` for variables) and eliminate redundant comments.

Reply:
I prefer to keep the math-sounding of "Ax=b". It is in linear solver, it's going to be about some Ax=b so I prefer A_, b_ and x_ (and it is shorter). Ok ?

Reply:
I tend to agree despite the fact it violates the member naming convention. I would get rid of the comments still since it is standard notation.